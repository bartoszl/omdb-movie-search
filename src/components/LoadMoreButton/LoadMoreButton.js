import styled from 'styled-components';

export default styled.button.attrs(() => ({
  type: 'button',
}))`
  color: white;
  font-size: 1.1rem;
  border: none;
  background-color: ${({ theme }) => theme.colors.background};
  cursor: pointer;
  padding: 0.75rem 1.5rem;
  box-shadow: 0 0 5px 1px ${({ theme }) => theme.colors.shadow};
  transition: all 0.25s;
  margin-top: 1.5rem;

  &:hover {
    box-shadow: 0 0 7px 2px ${({ theme }) => theme.colors.shadow};
  }
`;
