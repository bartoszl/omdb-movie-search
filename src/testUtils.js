import React from 'react';
import PropTypes from 'prop-types';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';
import theme from './config/theme';

const TestProviders = ({ children }) => (
  <ThemeProvider theme={theme}>
    <MemoryRouter>
      {children}
    </MemoryRouter>
  </ThemeProvider>
);

TestProviders.propTypes = {
  children: PropTypes.node.isRequired,
};

const customRender = (ui, options) => render(ui, { wrapper: TestProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
