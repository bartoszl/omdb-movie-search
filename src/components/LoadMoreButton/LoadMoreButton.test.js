import React from 'react';
import { render, cleanup } from '../../testUtils';
import LoadMoreButton from './LoadMoreButton';

describe('Loader', () => {
  afterEach(() => {
    cleanup();
  });

  test('matches snapshot', () => {
    const { asFragment } = render((
      <LoadMoreButton />
    ));

    expect(asFragment()).toMatchSnapshot();
  });
});
