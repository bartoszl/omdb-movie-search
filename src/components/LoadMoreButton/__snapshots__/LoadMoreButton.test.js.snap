// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="sc-bdnylx brcSWL"
    type="button"
  />
</DocumentFragment>
`;
